src/base/AnyMap.cpp

```diff
@@ -168,22 +235,241 @@ struct convert<Cantera::AnyMap> {
+    // the list
+    if (rhs.hasKey("__type__")) {
+        bool order_changed = false;
+        const auto& itemType = rhs["__type__"].asString();
+        std::unique_lock<std::mutex> lock(yaml_field_order_mutex);
+        if (AnyMap::s_headFields.count(itemType)) {
+            for (const auto& key : AnyMap::s_headFields[itemType]) {
+                for (auto& item : ordered) {
+                    if (std::get<0>(item).first >= 0) {
+                        // This and following items come from an input file and
+                        // should not be re-ordered
+                        break;
+                    }
+                    if (std::get<1>(item) == key) {
+                        std::get<0>(item).second = --head;
+                        order_changed = true;
+                    }
+                }
+            }
+        }
+        if (AnyMap::s_tailFields.count(itemType)) {
+            for (const auto& key : AnyMap::s_tailFields[itemType]) {
+                for (auto& item : ordered) {
+                    if (std::get<0>(item).first >= 0) {
+                        // This and following items come from an input file and
+                        // should not be re-ordered
+                        break;
+                    }
+                    if (std::get<1>(item) == key) {
+                        std::get<0>(item).second = ++tail;
+                        order_changed = true;
+                    }
+                }
+            }
+        }
+
+        if (order_changed) {
+            std::sort(ordered.begin(), ordered.end());
+        }
+    }
+
+    bool flow = rhs.getBool("__flow__", false);
+    if (flow) {
+        out << YAML::Flow;
+        out << YAML::BeginMap;
+        size_t width = 15;
+        for (const auto& item : ordered) {
+            const auto& name = std::get<1>(item);
+            const auto& value = *std::get<2>(item);
+            string valueStr;
+            bool foundType = true;
+            if (value.is<double>()) {
+                valueStr = formatDouble(value.asDouble(),
+                                     getPrecision(value));
+            } else if (value.is<string>()) {
+                valueStr = value.asString();
+            } else if (value.is<long int>()) {
+                valueStr = fmt::format("{}", value.asInt());
+            } else if (value.is<bool>()) {
+                valueStr = fmt::format("{}", value.asBool());
+            } else {
+                foundType = false;
+            }
+
+            if (foundType) {
+                if (width + name.size() + valueStr.size() + 4 > 79) {
+                    out << YAML::Newline;
+                    width = 15;
+                }
+                out << name;
+                out << valueStr;
+                width += name.size() + valueStr.size() + 4;
+            } else {
+                // Put items of an unknown (compound) type on a line alone
```

Comment:
As `AnyMap` is central to serialization, it may make sense to add a unit test for special cases (here and elsewhere as marked by CodeCov).